README.md

```diff
@@ -1,251 +1,220 @@
-```
-
-Launch a JSON API docker container: 
-
-```bash
-bin/start_jsonapi
-```
-
-Finally, run the tests:
-
-```
-npm test
+//Set up mongoose
+// JSON API URL
+const JSON_API_URI="http://localhost:8080/v1";
+//Stargate Coordinator Authentication URL
+const AUTH_URL="http://localhost:8081/v1/auth";
+// Define Product schema
+const productSchema = new Schema({ name: String, price: Number});
+// Create Product model
+const Product = mongoose.model('Product', productSchema);
+//Connect to server
+mongoose.connect(JSON_API_URI, {
+                    //Default user name
+                    "username":"cassandra",
+                    //Default password
+                    "password":"cassandra",
+                    "authUrl": AUTH_URL,
+                    //Name of the Namespace
+                    "keyspaceName": "inventory",
+                });
+//Wait for collections to get created
+Object.values(mongoose.connection.models).map(Model => Model.init());
+
+//Start the express server
+const HOST="0.0.0.0";
+const PORT=8097;
+const app = express();
+//Add product API
+app.get('/addproduct', (req, res) => {
+  const newProduct = new Product(
+    { name:"product"+Math.floor(Math.random() * 99 + 1), 
+      price: "" + Math.floor(Math.random() * 900 + 100) });
+  res.send('Added a product!');
+});
+//Get products API
+app.get('/getproducts', (req, res) => {
+  Product.find()
+    .then(products => res.json(products));
+});
+//Start server
+app.listen(PORT, HOST, () => {
+  console.log(`Running on http://${HOST}:${PORT}` 
+              + '\nNavigate to'
+              + '\n\thttp://localhost:'+PORT+'/addproducts'
+              + '\n\thttp://localhost:'+PORT+'/getproducts');
+});
 ```
+- Execute `node index.js`
+- Navigate to below urls
+    - http://localhost:8097/addproduct
+    - http://localhost:8097/getproducts
+
+## Architecture
+### High level architecture
+<img src="stargate-mongoose-arch.png" alt="stargate-mongoose usage end to end architecture"/>
+
+### Components
+- Cassandra Cluster - Apache Cassandra / DataStax Enterprise Cluster as backend database.
+- Stargate Coordinator Nodes - [Stargate](https://stargate.io/) is an opensource Data API Gateway for Cassandra. Coordinator is one of the primary components of Stargate which connects the API layer to the backend database. More details can be found [here](https://stargate.io/docs/latest/concepts/concepts.html#stargate-v2-0).
+- Stargate JSON API - [JSON API Server](https://github.com/stargate/jsonapi) is an opensource JSON API that runs on top of Stargate's coordinator.
+- JavaScript Clients that use mongoose - mongoose is an elegant mongodb object modeling library for node.js applications. By implementing a driver required by the mongoose interface to connect to the JSON API server instead of native mongodb access layer, now a JavaScript client can store/retrieve documents on an Apache Cassandra/DSE backend.
+
+## Version compatibility
+| Component/Library Name |    Version |
```

Comment:
won't you have to update this table whenever you update the file?

Maybe the way to do this is to specify the versions of mongoose and DSE (6.8.x) here, and then refer to the file for coordinator and JSON API.

Reply:
Nice idea, removed JSON API and Stargate from here.